flocker/node/agents/test/test_blockdevice.py

```diff
@@ -337,18 +338,15 @@ def assert_discovered_state(case,
             devices=expected_devices,
         ),
     )
-    if expected_nonmanifest_datasets is not None:
-        # FLOC-1806 - Make this actually be a dictionary (callers pass a list
-        # instead, despite the docs, and this is used like an iterable) and
-        # construct the ``NonManifestDatasets`` with the ``Dataset`` instances
-        # that are present as values.
-        expected += (
-            NonManifestDatasets(datasets={
-                unicode(dataset_id):
-                Dataset(dataset_id=unicode(dataset_id))
-                for dataset_id in expected_nonmanifest_datasets
-            }),
-        )
+    # FLOC-1806 - Make this actually be a dictionary (callers pass a list
+    # instead) and construct the ``NonManifestDatasets`` with the
+    # ``Dataset`` instances that are present as values.
+    expected += (
+        NonManifestDatasets(datasets={
+            unicode(dataset_id):
+            Dataset(dataset_id=unicode(dataset_id))
+            for dataset_id in expected_nonmanifest_datasets
+        }),)
     case.assertEqual(expected, state)
```

Comment:
I guess this helper would have been less fragile if it had been defined in terms of a `DeploymentState` instead of a sequence of changes.  Not actionable, just noting.